syntax = "proto3";

import "google/protobuf/empty.proto";

option go_package = ".;proto";


service Goods {
    // 商品接口
    rpc GoodsList (GoodsFilterRequest) returns (GoodsListResponse);
    // 用户提交订单有多个商品，批量查询商品信息
    rpc BatchGetGoods (BatchGoodsIdInfo) returns (GoodsListResponse);
    rpc CreateGoods (CreateGoodsInfo) returns (GoodsInfoResponse);
    rpc DeleteGoods (DeleteGoodsInfo) returns (google.protobuf.Empty);
    rpc UpdateGoods (CreateGoodsInfo) returns (google.protobuf.Empty);
    rpc GetGoodsDetail (GoodInfoRequest) returns (GoodsInfoResponse);
}


message GoodsFilterRequest {
    int32 price_min = 1;
    int32 price_max = 2;
    bool is_hot = 3;
    bool is_new = 4;
    bool is_tab = 5;
    int32 top_category = 6;
    // 页码
    uint32 page = 7;
    // 每页大小
    uint32 size = 8;
    string key_words = 9;
    int32 brand = 10;

}
message GoodsListResponse {
    int32 total = 1;
    repeated GoodsInfoResponse data = 2;
}

message BatchGoodsIdInfo {
    repeated int32 id = 1;
}

message CreateGoodsInfo {
    int32 id = 1;
    string name = 2;
    string goods_sn = 3;
    int32 stocks = 7; // 库存
    float market_price = 8;
    float shop_price = 9;
    string goods_brief = 10;
    string goods_desc = 11;
    bool ship_free = 12;
    repeated string images = 13;
    repeated string desc_images = 14;
    string goods_front_image = 15;
    bool is_new = 16;
    bool is_hot = 17;
    bool on_sale = 18;
    int32 category_id = 19;
    int32 brand_id = 20;
}

message CategoryBriefInfoResponse {
    int32 id = 1;
    string name = 2;
}

message GoodsInfoResponse {
    int32 id = 1;
    int32 category_id = 2;
    string name = 3;
    string goods_sn = 4;
    int32 click_num = 5;
    int32 sold_num = 6;
    int32 fav_num = 7;
    float market_price = 9;
    float shop_price = 10;
    string goods_brief = 11;
    string goods_desc = 12;
    bool ship_free = 13;
    repeated string images = 14;
    repeated string desc_images = 15;
    string goods_front_image = 16;
    bool is_new = 17;
    bool is_hot = 18;
    bool on_sale = 19;
    int64 add_time = 20;
    CategoryBriefInfoResponse category = 21;
    BrandInfoResponse brand = 22;
}


message DeleteGoodsInfo {
    int32 id = 1;
}


message GoodInfoRequest {
    int32 id = 1;
}


message BrandInfoResponse {
    int32 id = 1;
    string name = 2;
    string logo = 3;
}






